src/settings.cpp

```diff
@@ -70,5 +70,7 @@ void disableFlashPageRotation() { getEeprom().rotate(false); }
 
 void nukeSettings() {
   *Settings = DEFAULT_SETTINGS;
+  // set default value of the wifi power output
+  Settings->wifi_power = 9;
```

Comment:
One thing that came to my mind is that we need to sanity test this setting on load. When users install this new version and the old protocol buffer gets read, wifi_power will be initialized to 0. You have to handle that case and clamp the setting value to the same limits as the ones you have in the web form check

Reply:
Since im very new to Device programming and relatively rusty on C++ can you give me some guidance, where I can / should check the loaded Settings value for sanity (and probably reset it to a sane value)?
The nukeSettings is only called if  there are NO values in the eeprom or if somethig went wrong on decoding the eeprom value, right? So if old settings are loaded this won´t be called?
My intention would be putting the check and setting to a sane default value into the loadSettings() function, right before it returns after decoding into the Settings object. Right?

Reply:
Yeah for sure. Call the function sanitizeSettings() where you check wifi_power and clamp its values if necessary. Call it from both nukeSettings() instead of setting wifi_power directly and also from loadSettings() on line 43 in this file, right after "DPRINTF("Read and decoded settings, size = %d bytes.", len);"